src/libs/actions/IOU.ts

```diff
@@ -10743,6 +10748,285 @@ function getSearchOnyxUpdate({participant, transaction}: GetSearchOnyxUpdatePara
+                },
+                {
+                    transactionID: NumberUtils.rand64(),
+                    amount: Number(transactionDetails?.amount) / 2,
+                    description: transaction?.comment?.comment,
+                    category: transaction?.category,
+                    tags: transaction?.tag ? [transaction?.tag] : [],
+                    created: DateUtils.getDBTime(),
+                },
+            ],
+            amount: Number(transactionDetails?.amount),
+            currency: transactionDetails?.currency ?? CONST.CURRENCY.USD,
+            merchant: transactionDetails?.merchant ?? '',
+            participants: transaction?.participants,
+            attendees: transactionDetails?.attendees as Attendee[],
+            reportID,
+        },
+    });
+
+    Onyx.set(`${ONYXKEYS.COLLECTION.SPLIT_TRANSACTION_DRAFT}${transaction?.transactionID}`, draftTransaction);
+
+    Navigation.navigate(ROUTES.SPLIT_EXPENSE.getRoute(reportID ?? String(CONST.DEFAULT_NUMBER_ID), transaction.transactionID, undefined, Navigation.getActiveRouteWithoutParams()));
+}
+
+/**
+ * Add new split expense field
+ */
+function addSplitExpenseField(transaction: OnyxEntry<OnyxTypes.Transaction>, draftTransaction: OnyxEntry<OnyxTypes.Transaction>) {
+    if (!transaction || !draftTransaction) {
+        return;
+    }
+
+    Onyx.merge(`${ONYXKEYS.COLLECTION.SPLIT_TRANSACTION_DRAFT}${transaction.transactionID}`, {
+        comment: {
+            splitExpenses: [
+                ...(draftTransaction.comment?.splitExpenses ?? []),
+                {
+                    transactionID: NumberUtils.rand64(),
+                    amount: 0,
+                    description: transaction?.comment?.comment,
+                    category: transaction?.category,
+                    tags: transaction?.tag ? [transaction?.tag] : [],
+                    created: DateUtils.getDBTime(),
+                },
+            ],
+        },
+    });
+}
+
+/**
+ * Update split expense ammount
+ */
+function updateSplitExpenseAmountField(draftTransaction: OnyxEntry<OnyxTypes.Transaction>, currentItemTransactionID: string, amount: number) {
+    if (!draftTransaction?.transactionID || !currentItemTransactionID) {
+        return;
+    }
+
+    const updatedSplitExpenses = draftTransaction.comment?.splitExpenses?.map((splitExpense) => {
+        if (splitExpense.transactionID === currentItemTransactionID) {
+            return {
+                ...splitExpense,
+                amount,
+            };
+        }
+        return splitExpense;
+    });
+
+    Onyx.merge(`${ONYXKEYS.COLLECTION.SPLIT_TRANSACTION_DRAFT}${draftTransaction?.comment?.originalTransactionID}`, {
+        comment: {
+            splitExpenses: updatedSplitExpenses,
+        },
+    });
+}
+
+/**
+ * Complete split expense process
+ */
+function completeSplitTransaction(draftTransaction: OnyxEntry<OnyxTypes.Transaction>, hash: number) {
```

Comment:
Let's rename it to something more meaningful:

```suggestion
function saveSplitTransactions(draftTransaction: OnyxEntry<OnyxTypes.Transaction>, hash: number) {
```